### DISCUSSION TITLE
How to test nested fields with cel

### DISCUSSION THREAD

**Discussion by jsbrain** [cerbos/cerbos#1311] [Q&A] [answered]
So I'm trying to figure out how to express this:

```
has(R.attr.date_deleted) && (type(R.attr.date_deleted) != type(null))
```

so that it won't throw 

```
"error":"proto: invalid type: structpb.NullValue"
```

So date_deleted is either date/string or null. I want to ensure it's not null. But I always get some error and I really have no idea how to get this working ...

It worked in the REPL as I did something like `has(x.date_delete)` but now that I need to access two level deep it just won't work somehow.

**Comment by oguzhand95** [comment, marked as answer]
Hi @jsbrain,

I tried to produce what you've described in a playground instance;
https://play.cerbos.dev/p/gFbBDg8GSzInNB6169a1daskc3MgWMhe

Do you think this is what you are looking for? Please let me know if I misunderstood the problem.

**Reply by jsbrain** [reply]
Thanks @oguzhand95 for the quick response. 

So it seems the problems was not with my expression but with the `.planResource` function.

Ref: https://github.com/cerbos/cerbos/issues/1312
